cpp/modmesh/buffer/BufferExpander.hpp

```diff
@@ -127,6 +131,16 @@ class BufferExpander
         return data(it);
     }
 
+    using iterator = int8_t *;
+    using const_iterator = int8_t const *;
+
+    iterator begin() noexcept { return data(); }
```

Comment:
Use `m_begin`, `m_end` for iterator.

Reply:
I followed what `ConcreteBuffer` does, shall I also fix `ConcreteBuffer`?